src/Service/Startup.cs

```diff
@@ -445,7 +461,7 @@ public static ILoggerFactory CreateLoggerFactoryForHostedAndNonHostedScenario(IS
                 RuntimeConfigProvider configProvider = serviceProvider.GetRequiredService<RuntimeConfigProvider>();
                 if (configProvider.TryGetConfig(out RuntimeConfig? runtimeConfig))
                 {
-                    MinimumLogLevel = GetLogLevelBasedOnMode(runtimeConfig);
+                    MinimumLogLevel = GetLogLevelBasedOnLevel(runtimeConfig);
```

Comment:
We have a couple helper functions in startup now who are trying to get the log level based on values in the config, and the helper function here `GetLogLevelBasedOnLevel` depends on the other helper function in the case where the log level is null.

I think as Sean was pointing out in another comment that we should combine these into a GetLogLevel() function that sorts out all of the values and based on the precedence will return the correct log level. This function should probably live in the RuntimeConfig as Sean suggested, and in fact that previous helper function that we already had in startup probably should have been living in the RuntimeConfig to begin with.